### PR TITLE
MIR-959 derivate classification should have higher acl priority

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MIR-959).

### CODE REVIEW

File: mir-module/src/main/java/org/mycore/mir/authorization/MIRStrategy.java [hunk 1]
```diff
@@ -221,11 +221,12 @@ private boolean canEditPI() {
     }
 
     private Optional<MCRCategoryID> getAccessCategory(MCRObjectID objectId, MCRObjectID derivateId, String permission) {
-        String prefix = derivateId == null ? objectId.getTypeId() : derivateId.getTypeId();
+        final boolean isObject = derivateId == null;
+        String prefix = isObject ? objectId.getTypeId() : derivateId.getTypeId();
         List<MCRCategoryID> accessMappedCategories = getAccessMappedCategories(prefix, permission, accessClasses);
-        Optional<MCRCategoryID> accessCategory = getAccessCategory(accessMappedCategories, objectId)
+        return getAccessCategory(accessMappedCategories,
```
Comment (yagee-de) [inline mir-module/src/main/java/org/mycore/mir/authorization/MIRStrategy.java:227]: should be like
```java
return Optional.ofNullable(derivateId)
    .flatMap(id -> getAccessCategory(getAccessMappedCategories(id.getTypeId(), permission, accessClasses), id))
    .or(() -> getAccessCategory(getAccessMappedCategories(objectId.getTypeId(), permission, accessClasses), objectId));

```

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Review by kkrebs — APPROVED** [review]
successfully tested

File: mir-module/src/main/java/org/mycore/mir/authorization/MIRStrategy.java [hunk 2]
```diff
@@ -221,12 +221,8 @@ private boolean canEditPI() {
...
 
     private Optional<MCRCategoryID> getAccessCategory(MCRObjectID objectId, MCRObjectID derivateId, String permission) {
-        final boolean isObject = derivateId == null;
-        String prefix = isObject ? objectId.getTypeId() : derivateId.getTypeId();
-        List<MCRCategoryID> accessMappedCategories = getAccessMappedCategories(prefix, permission, accessClasses);
-        return getAccessCategory(accessMappedCategories,
-            isObject ? objectId : derivateId)
-            .or(() -> getAccessCategory(accessMappedCategories, derivateId));
+        final MCRObjectID id = Optional.ofNullable(derivateId).orElse(objectId);
+        return getAccessCategory(getAccessMappedCategories(id.getTypeId(), permission, accessClasses), id);
```
Comment (yagee-de) [inline mir-module/src/main/java/org/mycore/mir/authorization/MIRStrategy.java:225]: There is no fallback to object?

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with d69ddf87376dac215cc3008207a767f72b3cff4e! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/3125-pull_request-551

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 9db5f92fa7837a3c5adbda8e7561a4714b178156! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/3127-pull_request-551

**Review by yagee-de — APPROVED** [review]

**Pull request merged**